neqo-http3/src/connection.rs

```diff
@@ -1722,12 +1830,896 @@ mod tests {
         hconn.close(now(), 0, "");
     }
 
+    // Helper function
+    fn read_response(
+        mut hconn: Http3Connection,
+        mut neqo_trans_conn: Connection,
+        request_stream_id: u64,
+    ) {
+        let out = neqo_trans_conn.process(None, now());
+        hconn.process(out.dgram(), now());
+
+        let http_events = hconn.events();
+        for e in http_events {
+            match e {
+                Http3Event::HeaderReady { stream_id } => {
+                    assert_eq!(stream_id, request_stream_id);
+                    let h = hconn.read_response_headers(stream_id);
+                    assert_eq!(
+                        h,
+                        Ok((
+                            vec![
+                                (String::from(":status"), String::from("200")),
+                                (String::from("content-length"), String::from("3"))
+                            ],
+                            false
+                        ))
+                    );
+                }
+                Http3Event::DataReadable { stream_id } => {
+                    assert_eq!(stream_id, request_stream_id);
+                    let mut buf = [0u8; 100];
+                    let (amount, fin) = hconn
+                        .read_response_data(now(), stream_id, &mut buf)
+                        .unwrap();
+                    assert_eq!(fin, true);
+                    assert_eq!(amount, 3);
+                    assert_eq!(buf[..3], [0x61, 0x62, 0x63]);
```

Comment:
You should put the body in a constant so that you can replace all these constants with references.